util/src/main/scala/org/ensime/vfs/EnsimeVFS.scala

```diff
@@ -46,6 +46,22 @@ object `package` {
   private val zipFileField = classOf[ZipFileSystem].getDeclaredField("zipFile")
   zipFileField.setAccessible(true)
 
+  object URLParamEncoder {
+    def encode(input: String): String = {
```

Comment:
isn't that returning `String` or `Char`? From an efficiency point of view we create a lot of `URI`s so the current approach could contribute to quite a bit of churn. To really tackle that we might want to drop the `foreach` and use a `while` loop. But no need to prematurely optimise, let's see if it's a problem.